java-checks/src/main/java/org/sonar/java/checks/ForLoopFalseConditionCheck.java

```diff
@@ -90,14 +90,19 @@ private static boolean isConditionFalseAtInitialization(ForStatementTree forStat
 
   private static boolean evaluateCondition(ExpressionTree condition, int leftOperand, int rightOperand) {
     boolean conditionValue = true;
-    if (condition.is(Tree.Kind.GREATER_THAN)) {
-      conditionValue = leftOperand > rightOperand;
-    } else if (condition.is(Tree.Kind.GREATER_THAN_OR_EQUAL_TO)) {
-      conditionValue = leftOperand >= rightOperand;
-    } else if (condition.is(Tree.Kind.LESS_THAN)) {
-      conditionValue = leftOperand < rightOperand;
-    } else if (condition.is(Tree.Kind.LESS_THAN_OR_EQUAL_TO)) {
-      conditionValue = leftOperand <= rightOperand;
+    switch (condition.kind()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)